index.js

```diff
@@ -168,21 +180,26 @@ module.exports = class RandomAccessFile extends RandomAccessStorage {
       fs.rmdir(dir, onrmdir)
     }
   }
-}
 
-function readEmpty (req) {
-  req.callback(null, Buffer.alloc(0))
-}
+  _lock (mode, wait, cb) {
+    if (!fsctl) return cb(null)
```

Comment:
```suggestion
    if (!fsctl) return setImmediate(cb, null)
```

A callback function should never call back in the same tick (more cases below).

For example this demonstrates the problem:

```js
doSomething(() => {
  console.log('in callback')
})
doSomethingElse()
```

You would expect `doSomethingElse` to always happen before `"in callback"`.